src/net/sf/mpxj/primavera/PrimaveraPMFileReader.java

```diff
@@ -639,42 +639,45 @@ private void updateDates(Task parentTask)
          {
             updateDates(task);
 
-            if (actualStartDate == null || DateHelper.compare(actualStartDate, task.getActualStart()) > 0)
+            // the child tasks can have null dates (e.g. for nested wbs elements with no task children) so we
+            // still must protect against some children having null dates
+
+            if (actualStartDate == null || (task.getActualStart() != null && DateHelper.compare(actualStartDate, task.getActualStart()) > 0))
             {
                actualStartDate = task.getActualStart();
             }
 
-            if (actualFinishDate == null || DateHelper.compare(actualFinishDate, task.getActualFinish()) < 0)
+            if (actualFinishDate == null || (task.getActualFinish() != null && DateHelper.compare(actualFinishDate, task.getActualFinish()) < 0))
```

Comment:
Yes, you are correct that the null check isn't necessary for start dates and I can eliminate the unnecessary null check for start dates. However, it is actually needed for finish dates because `DateHelper.compare(actualDate, null) < 0 == true` so you can wind up incorrectly replacing valid finish dates with null unless you add the extra null check. Doing a quick code search, the same subtle bug appears to exist in the Phoenix reader code (if dates can be null from Phoenix). 

Whether you define null to be less than or greater than object instances, you're going to have an issue for either start or finish; do you think it's worth adding a null-safe `DateHelp.min/max` to fix and simplify this sort of code?